compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -109,8 +110,10 @@ trait TypeAssigner {
             if (tpe.isError) tpe
             else errorType(ex"$whatCanNot be accessed as a member of $pre$where.$whyNot", pos)
           }
-        }
-        else TypeOps.makePackageObjPrefixExplicit(tpe withDenot d)
+        else if ctx.compilationUnit != null && ctx.compilationUnit.isJava && tpe.isAnyRef then
```

Comment:
Are there scenarios where compilationUnit could be null? I am asking because we don't test this condition in any of the other 6 places where we test compilationUnit.isJava.

Reply:
> Are there scenarios where compilationUnit could be null? 

Yes, it happens in some tests but I haven't tried to track it down. The current situation is messy and I think we should have a default `NoCompilationUnit` unit instead of relying on null, but last time I looked into it this seemed like a non-trivial refactoring.

Reply:
As an intermediate step, we could add a method `isJava` to `Context` that does the safe compilationUnit lookup.